CrowsZA/functions/fn_surrenderChanceZeus.sqf

```diff
@@ -0,0 +1,81 @@
+/*/////////////////////////////////////////////////
+Author: Landric
+			   
+File: fn_surrenderChanceZeus.sqf
+Parameters: pos, unit
+Return: none
+
+*///////////////////////////////////////////////
+params [["_pos",[0,0,0],[[]],3], ["_unit",objNull,[objNull]]];
+
+private _onConfirm =
+{
+	params ["_dialogResult","_in"];
+	_in params [["_pos",[0,0,0],[[]],3], "_unit"];
+	private "_units";
+
+	if(isNull _unit) then {
+		_dialogResult params [
+			["_side", east, [east]],
+			["_confrontationRadius", 15, [15]],
+			["_surrenderChance", 0.33, [0.33]],
+			["_holdFire", true, [true]],
+			["_hesitation", 0, [0]]
+		];
+		_units = units _side;
+		[_units, _confrontationRadius, _surrenderChance, _holdFire, _hesitation] spawn {
+			params["_units", "_confrontationRadius", "_surrenderChance", "_holdFire", "_hesitation"];
+			{
+				if(!(isPlayer _x)) then {
+					[_x, _confrontationRadius, _surrenderChance, _holdFire, _hesitation] call crowsZA_fnc_surrenderChance;
```

Comment:
Not ideal you are calling the function inside scheduled space, when you in the function already spawns scheduled where needed. Might work fine, but scheduled space can be difficult at times, as you only get 2ms execution time, then you get paused and when working with units or similar something else can happen in that interval that changes how your code should be executed. 

Its not a big function, so could probably be fine. Otherwise you can use: https://cbateam.github.io/CBA_A3/docs/files/common/fnc_waitAndExecute-sqf.html 

instead of sleep, Will still execute it in unscheduled space.